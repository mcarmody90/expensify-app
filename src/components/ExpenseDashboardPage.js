import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExepsenseDashboardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExepsenseDashboardPage;