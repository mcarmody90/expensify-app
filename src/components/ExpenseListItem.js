import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, count, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{ description }</h3>
    </Link>
    <p>Amount: { (amount/100).toLocaleString("en-US", { style:"currency", currency:"USD" }) }</p>
    <p>Created: { createdAt }</p>
  </div>
);

export default ExpenseListItem;