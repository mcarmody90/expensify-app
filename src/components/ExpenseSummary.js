import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({ expenseCount, expenseTotal }) => (
  <div>
    {
      (expenseCount > 0)  && 
      (
        <p>
          Viewing {expenseCount} 
          {expenseCount === 1 ? ' expense ' : ' expenses '}
          totalling: {numeral(expenseTotal / 100).format('$0,0.00')}.
        </p>
      )
    }
  </div>
);

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expenseTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);