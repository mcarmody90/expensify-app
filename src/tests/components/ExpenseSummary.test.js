import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary with no expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={0} expenseTotal={999} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseList with expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={3} expenseTotal={20} />);
  expect(wrapper).toMatchSnapshot();
});