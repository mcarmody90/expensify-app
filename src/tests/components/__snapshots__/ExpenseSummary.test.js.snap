// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     expenses 
    totalling: 
    $0.20
    .
  </p>
</div>
`;

exports[`should render ExpenseSummary with no expenses 1`] = `<div />`;
